pkg/installer/custominstallconfig.go

```diff
@@ -203,3 +211,28 @@ func appendFilesToCvoOverrides(a asset.WritableAsset, g graph.Graph) (err error)
 
 	return nil
 }
+
+// replacePointerIgnition performs the same functionality as the upstream
+// installer's pointerIgnitionConfig() but with ARO specific DNS config
+func replacePointerIgnition(a asset.WritableAsset, g graph.Graph, aroDNSConfig *bootkube.ARODNSConfig) (err error) {
+	masterPointerIgn := g.Get(&machine.Master{}).(*machine.Master)
+	ignitionHost := net.JoinHostPort(aroDNSConfig.APIIntIP, "22623")
+	role := "master"
+
+	masterPointerIgn.Config.Ignition.Config.Merge[0].Source = util.StrToPtr(func() *url.URL {
+		return &url.URL{
+			Scheme: "https",
+			Host:   ignitionHost,
+			Path:   fmt.Sprintf("/config/%s", role),
```

Comment:
Is there any value in getting everything but the host from parsing the existing URL?